back/engines/commercial/bulk_import_ideas/app/controllers/bulk_import_ideas/web_api/v1/bulk_import_controller.rb

```diff
@@ -126,13 +134,21 @@ def sidefx
 
     def importer_service
       model = params[:model]
-      CONSTANTIZER.fetch(model)[:importer_class]
+      locale = params[:import] ? bulk_create_params[:locale] : current_user.locale
+
+      service = CONSTANTIZER.fetch(model)[:importer_class]
+      @importer_service ||= service.new(current_user, locale)
     end
 
     def file_parser_service
       model = params[:model]
       format = params[:format]
-      CONSTANTIZER.fetch(model).fetch(format)[:parser_class]
+      locale = params[:import] ? bulk_create_params[:locale] : current_user.locale
+      personal_data_enabled = params[:import] ? bulk_create_params[:personal_data] || false : false
```

Comment:
```suggestion
      personal_data_enabled = params[:import].present? && bulk_create_params[:personal_data].present?
```
Is it the same? If so, maybe (not sure!) it's a bit clearer.